app/src/pages/prompt/promptCodeSnippets.tsx

```diff
@@ -0,0 +1,223 @@
+import { CodeLanguage } from "@phoenix/components/code";
+
+import type { PromptCodeExportCard__main$data as PromptVersion } from "./__generated__/PromptCodeExportCard__main.graphql";
+
+export type PromptToSnippet = ({
+  invocationParameters,
+  modelName,
+  modelProvider,
+  outputSchema,
+  tools,
+  template,
+}: Pick<
+  PromptVersion,
+  | "invocationParameters"
+  | "modelName"
+  | "modelProvider"
+  | "outputSchema"
+  | "tools"
+  | "template"
+>) => string;
+
+const TABS = "  ";
+
+/**
+ * Indentation-aware JSON formatter
+ *
+ * @returns The formatted JSON string
+ */
+const jsonFormatter = ({
+  json,
+  level,
+  removeKeyQuotes = false,
+}: {
+  /**
+   * The JSON object to format
+   */
+  json: unknown;
+  /**
+   * The indentation level
+   */
+  level: number;
+  /**
+   * Whether to remove quotes from keys
+   */
+  removeKeyQuotes?: boolean;
+}) => {
+  const tabsWithLevel = TABS.repeat(level);
+  let fmt = JSON.stringify(json, null, TABS);
+  // add TABS before every line except the first
+  // this allows you to add additional indentation to the emitted JSON
+  fmt = fmt
+    .split("\n")
+    .map((line, index) => (index > 0 ? tabsWithLevel + line : line))
+    .join("\n");
+
+  if (removeKeyQuotes) {
+    // Replace quoted keys with unquoted keys, but only when they're valid identifiers
+    fmt = fmt.replace(/"([a-zA-Z_$][a-zA-Z0-9_$]*)"\s*:/g, "$1:");
+  }
+
+  return fmt;
+};
+
+/**
+ * A map of languages to model providers to code snippets
+ *
+ * @todo replace with a react-like DSL, for example something like the following:
+ * @example
+ * ```tsx
+ * code(
+ *   { language, provider },
+ *   [
+ *     openai(),
+ *     messages({messages}),
+ *     completion(null, [argument(null, tools({ tools }))
+ *   ]
+ * )
+ * ```
+ * where each function takes a props object and optional children, and returns a string.
+ *
+ * That way, each component can manage how to emit its portion of the string based on language and model provider,
+ * accessible via context from the top level code component.
```

Comment:
I think I want to take a stab at this, but in a follow-up PR.

This would make it much easier to maintain code samples across the matrix of code language and model provider. I think 😅 ...

Reply:
I wouldn't optimize this flow quite yet - while it's tempting think there's probably language specifics that just aren't generic enough to warent building this right now.